### DISCUSSION TITLE
Import Menu is undefined in render script

### DISCUSSION THREAD

**Discussion by surajvofox** [cawa-93/vite-electron-builder#457] [Q&A]
**Describe the bug**
Platform(Mac os)

I tried to access the Electron Menu in the render script.I used the preload for that. so i tried to import the menu from the electron api. but it was empty when i log it with in the preload script itself.
Next i tired to enable the remoteModule in the webprefrence, and tried to access the menu in the preload from remote.when i console it it showing the menu init, but when i tried to pass that to render script and console it in the render, its still don't have the menu . I tried to console log remote in the render script but it has only minimum methods only. and Menu was not there.

Below is the image which shows when i console. the remote. from render script.

![image](https://user-images.githubusercontent.com/58926728/134803114-390a8630-ffb1-4953-be8d-9f994b4f9e8b.png)

Below is the image that i consoled from the preload script which consist all the methods , when i tried to pass the menu  only from the preload script to render script still its not working. showing menu is undefined

![image](https://user-images.githubusercontent.com/58926728/134803215-d7571c20-d392-4dd5-9b46-1fd34dab6a9b.png)

**Comment by cawa-93** [comment]
Preload script give you access to node.js api (like `require` or resolving `node_module`), but it still part of __Renderer__ process. But Menu API available only in __Main__ process ([see docs](https://www.electronjs.org/docs/latest/api/menu/)).

So, you need to send message to Main process and call Menu there. [See example in docs](https://www.electronjs.org/docs/latest/api/menu/#render-process).
